### ISSUE TITLE
Weird typing of NamedTuple `index` field

### ISSUE THREAD

**Issue by anxolerd** [python/mypy#4507]
# Note
I am not sure whether it should be reported here or to the typeshed repo. Reporting here for now. If this is a typeshed issue, I'll gladly open issue there and close this one.

If there is something special in `index` field in `NamedTuple`, please point out to the corresponding documentation as I failed to find anything on topic.

# Description

## Type
bug

## Code to check

```
from typing import NamedTuple


class Foo(NamedTuple):
    index: str
```

## Actual behavior
```
test.py:5: error: Incompatible types in assignment (expression has type "str", base class "tuple" defined the type as "Callable[[Tuple[str, ...], Any, int, int], int]")
```

## Expected behavior
No errors while typechecking

## Environment
```
$ mypy -V
mypy 0.560
$ python -V
Python 3.6.4
```

Behavior is still the same when using mypy from master:

```
$ mypy -V
mypy 0.570-dev-7192a7578fc47530bbb5382984d289bffa72b30b
$ mypy test.py
test.py:5: error: Incompatible types in assignment (expression has type "str", base class "tuple" defined the type as "Callable[[Tuple[str, ...], Any, int, int], int]")
```

**Comment by ethanhs** [comment]
Agreed. Callable types in base classes should probably have their own special clearer message.

**Comment by PeterJCLaw** [comment]
(Cross posted from https://github.com/python/mypy/issues/3950#issuecomment-836623894)

My experience was that I didn't initially understand that the assignment issue was to do with the assignment of the member (in my case `count`) into the class. It felt like I was _declaring_ a class member, which feels conceptually different (even if it's technically an assignment).

Would an error message like the following be possible here (emphasis to show changes):
> Incompatible types in _class member declaration_ (expression has type "int", base class "tuple" defined the _member_ type as "Callable[[Tuple[int, ...], Any], int]") [assignment]

Maybe even add something like:
> Hint: are you overriding a method with an attribute?

If there's scope to change the error code from `assignment` to perhaps `member-declaration` (or something) that might also help? Not sure if that would be too breaking though.

For comparison, the method override error message is (to me) much clearer:
> Signature of "count" incompatible with supertype "tuple"  [override]
